src/main/java/software/amazon/event/kafkaconnector/batch/BestEffortInOrderEventBridgeBatching.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package software.amazon.event.kafkaconnector.batch;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collector;
+import java.util.stream.Stream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.services.eventbridge.model.PutEventsRequestEntry;
+import software.amazon.event.kafkaconnector.util.MappedSinkRecord;
+
+public class BestEffortInOrderEventBridgeBatching implements EventBridgeBatchingStrategy {
+
+  private static final Logger logger =
+      LoggerFactory.getLogger(BestEffortInOrderEventBridgeBatching.class);
+
+  private final Collector<
+          MappedSinkRecord<PutEventsRequestEntry>,
+          PutEventsRequestEntrySizeCollector,
+          Stream<List<MappedSinkRecord<PutEventsRequestEntry>>>>
+      collector =
+          Collector.of(
+              PutEventsRequestEntrySizeCollector::new,
+              PutEventsRequestEntrySizeCollector::accumulate,
+              (left, right) -> {
+                throw new IllegalStateException();
+              },
+              PutEventsRequestEntrySizeCollector::finish);
+
+  @Override
+  public Stream<List<MappedSinkRecord<PutEventsRequestEntry>>> apply(
+      Stream<MappedSinkRecord<PutEventsRequestEntry>> records) {
+    if (records.isParallel()) {
+      throw new IllegalArgumentException("Stream must not be parallel.");
+    }
+    return records.collect(collector);
+  }
+
+  static class PutEventsRequestEntrySizeCollector {
+
+    private static final int MAX_BATCH_SIZE_BYTES = 256 * 1024;
+
+    private final List<List<MappedSinkRecord<PutEventsRequestEntry>>> batches = new ArrayList<>();
+    private int actualBatchSize = 0;
+    private int index = 0;
+
+    PutEventsRequestEntrySizeCollector() {
+      batches.add(new ArrayList<>());
+    }
+
+    void accumulate(final MappedSinkRecord<PutEventsRequestEntry> item) {
+
+      var itemSize = getSize(item.getValue());
+      if (itemSize > MAX_BATCH_SIZE_BYTES) {
+        logger.warn(
+            "Item for {} exceeds EventBridge size limit. Size is {} bytes.",
+            item.getSinkRecord(),
```

Comment:
+1 on this! This would put that record in a separate (single) batch which would fail during `PutEvents` as discussed, not retried and logged/discarded/DLQ-ed. Can we please get a test for this behavior (using mock EventBridge response)?

Note: this implementation might reveal sensitive data. IIUC, in other places where we log we mostly try to only log topic/partition/offset.

Reply:
I will restrict the logging content to topic/partition/offset and add another test for `EventBridgeWriter` soon.